proxy/coco/src/project/create/validation.rs

```diff
@@ -173,8 +173,8 @@ impl Repository {
                 let repo = git2::Repository::open(path.clone())
                     .or_matches(git_ext::is_not_found_err, || Err(Error::NotARepo(path)))?;
 
-                let _ = Self::existing_branch(&repo, &default_branch)?;
-                let _ = Self::existing_remote(&repo, &url)?;
+                Self::existing_branch(&repo, &default_branch)?;
```

Comment:
Is this part of clippy/fmt? I tend to prefer knowing when a value that isn't `()` is returned but is being ignored. But that's a personal preference and not a hill I'll die on :)

Reply:
I second this sentiment.

Reply:
Clippy [complains](https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_drop) because the `git2::Refernce` implements `Drop`. Alternatives for fixing this are explicit variable names
~~~
let _default_branch_ref = Self::existing_branch(&repo, &default_branch)?;
~~~
explicit drop
~~~
drop(Self::existing_branch(&repo, &default_branch)?);
~~~
or disabling this lint.

I have no preference either way and would leave it up to you.

Reply:
Cool, thanks for the link! I think the `_name` would be my preferred convention.

Reply:
> Cool, thanks for the link! I think the `_name` would be my preferred convention.

Well, that doesn’t work quite as expected:

~~~rust
let _remote = Self::existing_remote(&repo, &url)?;
Ok(Self::Existing {
    repo,
    url,
    default_branch,
})
~~~

The borrow checker complains because the lifetime of `_remote` is the same as the lifetime of `&repo`, that is the whole scope. But `repo` is moved in the final statement. I’m leaning towards just ignoring this lint. Alternatively we could create a dedicated scope around the two `Self::existing_remote` statements or use an explicit `drop`.

Reply:
I would probably opt for introducing a scope, personally. Sorry, this is a pain :sweat_smile:

Reply:
No worries. It’s fixed in https://github.com/radicle-dev/radicle-upstream/pull/1548/commits/29b0e54e8a4bc63c482c1ba1d155fff6a314beb6